ee/src/main/java/org/jboss/as/ee/component/deployers/InterceptorAnnotationProcessor.java

```diff
@@ -82,13 +83,14 @@ private void processComponentConfig(final EEApplicationClasses applicationClasse
         final Class<?> componentClass;
         try {
             componentClass = ClassLoadingUtils.loadClass(description.getComponentClassName(), deploymentUnit);
+            handleAnnotations(applicationClasses, deploymentReflectionIndex, componentClass, description);
         } catch (Throwable e) {
             //just ignore the class for now.
             //if it is an optional component this is ok, if it is not an optional component
             //it will fail at configure time anyway
+            EeLogger.ROOT_LOGGER.debugf(e,"Ignoring failure to handle interceptor annotations for %s", description.getComponentClassName());
             return;
         }
-        handleAnnotations(applicationClasses, deploymentReflectionIndex, componentClass, description);
     }
```

Comment:
LGTM. Some minor points: we can now remove `throws DeploymentUnitProcessingException` since the method already catches all exceptions. `return;` is now also redundant. `final <class> componentClass` can be combined with assignment.

Reply:
Thanks; I cleaned those things up.